src/Security/test/Authorization.Certificate.Test/CertificateHttpClientBuilderExtensionsTest.cs

```diff
@@ -31,11 +31,40 @@ public async Task AddCertificateAuthorizationClient_AddsNamedHttpClientWithCerti
         certificateHeader.Should().BeEquivalentTo(Convert.ToBase64String(certificateBytes));
     }
 
-    private static IHostBuilder GetHostBuilder()
+    [Fact]
+    public async Task AddCertificateAuthorizationClient_AllowsCustomHeader()
+    {
+        const string customCertificateHeader = "my-arbitrary-header";
+        byte[] certificateBytes = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key").Export(X509ContentType.Cert);
+        using var appScope = new EnvironmentVariableScope("VCAP_APPLICATION", "not empty");
+        using var certScope = new EnvironmentVariableScope("CF_INSTANCE_CERT", "instance.crt");
+        using var keyScope = new EnvironmentVariableScope("CF_INSTANCE_KEY", "instance.key");
+        using IHost host = await GetHostBuilder(customCertificateHeader).StartAsync();
+        var factory = host.Services.GetRequiredService<IHttpClientFactory>();
+        HttpClient client = factory.CreateClient("test");
+
+        client.Should().NotBeNull();
+        client.DefaultRequestHeaders.Contains(customCertificateHeader).Should().BeTrue();
+        string certificateHeader = client.DefaultRequestHeaders.GetValues(customCertificateHeader).First();
+        certificateHeader.Should().BeEquivalentTo(Convert.ToBase64String(certificateBytes));
```

Comment:
```suggestion
        certificateHeader.Should().Be(Convert.ToBase64String(certificateBytes));
```
Please try to stay away from `BeEquivalentTo`, which uses a complex reflection-based recursive object graph comparison.